subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dependencies/DefaultResolvedVersionConstraint.java

```diff
@@ -45,10 +46,16 @@ public DefaultResolvedVersionConstraint(String requiredVersion, String preferred
         this.preferredVersionSelector = preferredVersion.isEmpty() ? null : scheme.parseSelector(preferredVersion);
 
         if (strict) {
+            //if (!rejectedVersions.isEmpty()) {
+            //    throw new IllegalArgumentException("Cannot combine 'strict' and 'reject' in a single version constraint.");
+            //}
```

Comment:
Can you remove this dead code OR add/replace it with a comment?

Reply:
removed